### PR TITLE
Feature/upse 103

### LINKED ISSUE #72: RomeAdapter uses deprecated HTTPClient APIs that prevent it from establishing some SSL connections
Here is an issue ticket from the Eureka project that seems to capture our situation well...

  - https://github.com/Netflix/eureka/issues/873

Here is the place in the Spring context where we create the `httpClient` bean...

  - https://github.com/Jasig/NewsReaderPortlet/blob/master/src/main/resources/context/applicationContext.xml#L87

`DefaultHttpClient` is deprecated;  we're meant to use `HttpClientBuilder`.

### PR DESCRIPTION
**Pull request by mgillian**
Updates httpClient to current version, resolves error that prevents establishing some SSL connections 

Resolves #72

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -170,7 +170,7 @@
             <!-- for cernunnos -->
                 <groupId>javax.script</groupId>
                 <artifactId>groovy-engine</artifactId>
-                <version>1.1</version>
+                <version>1.0</version>
```
Comment (ChristianMurphy) [inline pom.xml:173]: do we need to downgrade the groovy engine?
is it causing an issue with cernunnos?

**Comment by mgillian** [comment]
This downgrade was because the version wasn't available from a Maven
repository.  If the repository is available again, we can upgrade it
again.  I haven't checked in the past couple of weeks.

Michael

**Comment by ChristianMurphy** [comment]
> This downgrade was because the version wasn't available from a Maven repository. If the repository is available again, we can upgrade it again. I haven't checked in the past couple of weeks.

It might be worth looking into that.
It looks like as of 5 days ago, the CI/CD server was able to find all the repositories https://github.com/uPortal-Project/NewsReaderPortlet/actions/runs/2683677067
Which, in theory, should include this one.

If it is back, it would be good to keep the version updated.
If the version is still missing, it may be worth looking into what else CI/CD should check to be aware of this dependency.

**Comment by mgillian** [comment]
Moved groovy-engine back to 1.1 and tried to build:

[INFO] ----------------< org.jasig.portlet:NewsReaderPortlet
>-----------------
[INFO] Building News Portlet 5.0.5-SNAPSHOT
[INFO] --------------------------------[ war
]---------------------------------
[WARNING] The POM for javax.script:groovy-engine:jar:1.1 is missing, no
dependency information available
[INFO]
------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO]
------------------------------------------------------------------------
[INFO] Total time:  0.484 s
[INFO] Finished at: 2022-07-21T14:13:12-07:00
[INFO]
------------------------------------------------------------------------
[ERROR] Failed to execute goal on project NewsReaderPortlet: Could not
resolve dependencies for project
org.jasig.portlet:NewsReaderPortlet:war:5.0.5-SNAPSHOT: Failure to find
javax.script:groovy-engine:jar:1.1 in
https://repository.mulesoft.org/nexus/content/repositories/public/ was
cached in the local repository, resolution will not be reattempted until
the update interval of Mulesoft Repository has elapsed or updates are
forced -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e
switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions,
please read the following articles:
[ERROR] [Help 1]
http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
***@***.***:~/customers/uportal/git/NewsReaderPortlet$

I'm open to suggestions on how to troubleshoot.  If it's something in my
local environment, that would make sense.  Are we still able to build/test
based on what is in GitHub?

Michael

**Comment by ChristianMurphy** [comment]
I tried the following:

```bash
# reset maven cache
rm -rf ~/.m2/repository/

# check java version
java -version
# openjdk version "1.8.0_332"
# OpenJDK Runtime Environment Corretto-8.332.08.1 (build 1.8.0_332-b08)
# OpenJDK 64-Bit Server VM Corretto-8.332.08.1 (build 25.332-b08, mixed mode)

# check maven version
mvn -version
# Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
# Maven home: /home/developer/.sdkman/candidates/maven/current
# Java version: 1.8.0_332, vendor: Amazon.com Inc., runtime: /home/developer/.sdkman/candidates/java/8.332.08.1-amzn/jre

# install and build project
mvn install
```

and don't see the issue described above.

A few debugging ideas:
* what maven version are you on? Is it version 3?
* what java version are you running?
* are you running a VPN or through a internet gateway that may block maven central connections?

**Comment by ChristianMurphy** [comment]
> Are we still able to build/test based on what is in GitHub?

Yes, both the different versions of the code in version control can be accessed, and GitHub actions CI/CD can be used for testing cross platform (these are the results from the build on this branch https://github.com/uPortal-Project/NewsReaderPortlet/actions/runs/2714494235, further results can be viewed/accessed from the "checks" tab in the GitHub pull request UI)

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**